src/full_coverage_path_planner.cpp

```diff
@@ -32,218 +32,217 @@
-{
-  tf2::Quaternion q;
-  q.setRPY(0, 0, yaw);
-  return tf2::toMsg(q);
-}
-
-void FullCoveragePathPlanner::publishPlan(const std::vector<geometry_msgs::msg::PoseStamped>& path)
-{
-  if (!initialized_)
+  FullCoveragePathPlanner::FullCoveragePathPlanner() : initialized_(false)
   {
-    RCLCPP_ERROR(rclcpp::get_logger("FullCoveragePathPlanner"), "This planner has not been initialized yet, but it is being used, please call initialize() before use");
-    return;
   }
 
-  // create a message for the plan
-  nav_msgs::msg::Path gui_path;
-  gui_path.poses.resize(path.size());
-
-  if (!path.empty())
+  auto FullCoveragePathPlanner::createQuaternionMsgFromYaw(double yaw)
   {
-    gui_path.header.frame_id = path[0].header.frame_id;
-    gui_path.header.stamp = path[0].header.stamp;
+    tf2::Quaternion q;
+    q.setRPY(0, 0, yaw);
+    return tf2::toMsg(q);
   }
 
-  // Extract the plan in world co-ordinates, we assume the path is all in the same frame
-  for (unsigned int i = 0; i < path.size(); i++)
+  void FullCoveragePathPlanner::publishPlan(const std::vector<geometry_msgs::msg::PoseStamped> &path)
   {
-    gui_path.poses[i] = path[i];
-  }
+    if (!initialized_)
+    {
+      RCLCPP_ERROR(rclcpp::get_logger("FullCoveragePathPlanner"), "This planner has not been initialized yet, but it is being used, please call initialize() before use");
+      return;
+    }
 
-  plan_pub_->publish(gui_path);
-}
+    // create a message for the plan
+    nav_msgs::msg::Path gui_path;
+    gui_path.poses.resize(path.size());
 
-void FullCoveragePathPlanner::parsePointlist2Plan(const geometry_msgs::msg::PoseStamped& start,
-    std::list<Point_t> const& goalpoints,
-    std::vector<geometry_msgs::msg::PoseStamped>& plan)
-{
-  geometry_msgs::msg::PoseStamped new_goal;
-  std::list<Point_t>::const_iterator it, it_next, it_prev;
-  int dx_now, dy_now, dx_next, dy_next, move_dir_now = 0, move_dir_next = 0;
-  bool do_publish = false;
-  float orientation = eDirNone;
-  RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Received goalpoints with length: %lu", goalpoints.size());
-  if (goalpoints.size() > 1)
-  {
-    for (it = goalpoints.begin(); it != goalpoints.end(); ++it)
+    if (!path.empty())
     {
-      it_next = it;
-      it_next++;
-      it_prev = it;
-      it_prev--;
+      gui_path.header.frame_id = path[0].header.frame_id;
+      gui_path.header.stamp = path[0].header.stamp;
+    }
 
-      // Check for the direction of movement
-      if (it == goalpoints.begin())
-      {
-        dx_now = it_next->x - it->x;
-        dy_now = it_next->y - it->y;
-      }
-      else
-      {
-        dx_now = it->x - it_prev->x;
-        dy_now = it->y - it_prev->y;
-        dx_next = it_next->x - it->x;
-        dy_next = it_next->y - it->y;
-      }
+    // Extract the plan in world co-ordinates, we assume the path is all in the same frame
+    for (unsigned int i = 0; i < path.size(); i++)
+    {
+      gui_path.poses[i] = path[i];
+    }
```

Comment:
Simply declare the gui_path here for a more limited scope throughout your code

```suggestion
    nav_msgs::msg::Path gui_path;
    gui_path.header.frame_id = path[0].header.frame_id;
    gui_path.header.stamp = path[0].header.stamp;
    for (const auto pose : path)
    {
      gui_path.poses.append(pose);
    }
    plan_pub_->publish(gui_path);
```